### PR TITLE
chore: Dex integration 

### PR DESCRIPTION
**Pull request by jy4096**
configurable Dex auth

### CODE REVIEW

File: config/base/dex/numaflow-dex-secrets.yaml [hunk 1]
```diff
@@ -3,5 +3,5 @@ kind: Secret
 metadata:
   name: numaflow-dex-secrets
 stringData:
-  dex-github-client-id: c1812606eaf8c6e53a79
-  dex-github-client-secret: c5244582cb71f9e341968fc9bcd72edec3c5aebc
\ No newline at end of file
+  dex-github-client-id: bfa31f09c135d86c77c5
```
Comment (whynowy) [inline config/base/dex/numaflow-dex-secrets.yaml:6]: Is it able to make these 2 empty?

File: config/namespace-install.yaml [hunk 2]
```diff
@@ -16299,8 +16299,8 @@ kind: Secret
 metadata:
   name: numaflow-dex-secrets
 stringData:
-  dex-github-client-id: c1812606eaf8c6e53a79
-  dex-github-client-secret: c5244582cb71f9e341968fc9bcd72edec3c5aebc
+  dex-github-client-id: bfa31f09c135d86c77c5
+  dex-github-client-secret: 6c5444bcd246d514ea7b41033b8a5e3490fc8adc
```
Comment (vigith) [inline config/namespace-install.yaml:16303]: whose client-id is this?

File: config/base/dex/numaflow-dex-secrets.yaml (hunk 1 above)
Comment (KeranYang) [inline config/base/dex/numaflow-dex-secrets.yaml:6]: where do we get the ID and secret string? could you add a description for it?

File: config/base/dex/numaflow-dex-server-configmap.yaml [hunk 3]
```diff
@@ -4,29 +4,28 @@ metadata:
   name: numaflow-dex-server
 data:
   config.yaml: |
-    issuer: hostname/dex
+    issuer: https://localhost:8443/dex
```
Comment (whynowy) [inline config/base/dex/numaflow-dex-server-configmap.yaml:7]: This hostname seems to be weird.

File: config/base/dex/numaflow-dex-server-deployment.yaml [hunk 4]
```diff
@@ -10,50 +10,34 @@ spec:
   selector:
     matchLabels:
       app.kubernetes.io/name: numaflow-dex-server
-      app.kubernetes.io/part-of: numaflow
```
Comment (whynowy) [inline config/base/dex/numaflow-dex-server-deployment.yaml:13]: Can we get these 2 labels back?

**Review by KeranYang — APPROVED** [review]

File: cmd/commands/server.go [hunk 5]
```diff
@@ -65,6 +69,8 @@ func NewServerCommand() *cobra.Command {
 	command.Flags().StringVar(&managedNamespace, "managed-namespace", sharedutil.LookupEnvStringOr("NAMESPACE", "numaflow-system"), "The namespace that the server watches when \"--namespaced\" is \"true\".")
 	command.Flags().StringVar(&baseHref, "base-href", "/", "Base href for Numaflow server, defaults to '/'.")
 	command.Flags().BoolVar(&disableAuth, "disable-auth", false, "Whether to disable authentication and authorization, defaults to false.")
-	command.Flags().StringVar(&dexServerAddr, "dex-server-addr", "http://numaflow-dex-server:5556", "The address of the Dex server.")
+	command.Flags().StringVar(&dexServerAddr, "dex-server-addr", "http://numaflow-dex-server:5556/dex", "The actual address of the Dex server for the reverse proxy to target.")
+	command.Flags().StringVar(&dexProxyAddr, "dex-proxy-addr", "https://localhost:8443/dex", "The proxy address of the Dex server.")
```
Comment (whynowy) [inline cmd/commands/server.go:73]: Can we get the proxy address from the server-addr?

Comment (jy4096) [reply cmd/commands/server.go:73]: If serverAddr is localhost then proxy is also localhost... otherwise proxy will be `https://numaflow-server:8443/dex`

**Pull request merged**